lib/actions/reassemble.js

```diff
@@ -127,9 +114,194 @@ async function processAction(msg) {
       groupElements.push({groupSize: messagesNumberSeen, groupId: groupId, messageData:incomingData})
     }
   }
-  if(messageGroupSize === undefined && msg.body.timersec === undefined){
-    throw new Error('Either group size or delay timer need to be defined');
+
+  // When both groupSize and delay timer option is selected 
+  if(mode == 'groupSize&timeout'){ 
+
+    var delay = msg.body.timersec
+    if(delay >= 40000){delay == 40000}
+    clearTimeout(timeHandle);
+    timeHandle = setTimeout(timer, delay, this); 
+
+    if(groupList.includes(groupId)){
+      parsedMessageGroup.messages.forEach((message) => {
+        incomingData[message.messageId] = message.messageData;
+      });
+      var currentGroup = groupElements.filter(function(x) { return x.groupId == groupId; }); // update incoming messageData in current group
+      currentGroup[0].groupSize = messagesNumberSeen
+      currentGroup[0].messageData = incomingData
+      groupElements = groupElements.filter(function(x) { return x.groupId != groupId; });
+      groupElements.push(currentGroup[0]);
+    }
+    else{
+      parsedMessageGroup.messages.forEach((message) => {
+        incomingData[message.messageId] = message.messageData;
+      });
+      groupList.push(groupId)
+      groupElements.push({groupSize: messagesNumberSeen, groupId: groupId, messageData:incomingData})
+    }
+    if (messagesNumberSeen >= messageGroupSize) {
+      parsedMessageGroup.messages.forEach((message) => {
+        incomingData[message.messageId] = message.messageData;
+      });
+  
+      await this.emit('data', messages.newMessageWithBody({
+        groupSize,
+        groupId,
+        messageData: incomingData,
+      }));
+      await storage.deleteObjectById(messageGroupId);
+      this.logger.info(`Message group with id ${messageGroupId} has been deleted`);
+      groupList = groupList.filter(def => def != groupId);
+      groupElements = groupElements.filter(def => def.groupId != groupId);
+    }
   }
 }
 
-exports.process = processAction;
+//----------------------------------------------------------------------------------------------------------------------------
+//Dynamic drop-down logic starts hier
+
+async function dynList(cfg) {
+  let list = {
+      "groupSize": "Use Group Size",
+      "timeout": "Use Timeout",
+      "groupSize&timeout": "Use Group Size and Timeout"
+  };
+  return list;
```

Comment:
It's possible to configure a static dropdown directly in `component.json`

Reply:
Done!